src/rules/requests/request-handlers.ts

```diff
@@ -597,16 +597,18 @@ export class PassThroughHandler extends PassThroughHandlerDefinition {
             reqUrl = modifiedReq?.url || reqUrl;
 
             headersManuallyModified = !!modifiedReq?.headers;
-            let headers = modifiedReq?.headers || clientReq.headers;
-
+            const clientHeaders = rawHeadersToObject(clientReq.rawHeaders)
+            let headers = modifiedReq?.headers || clientHeaders;
             Object.assign(headers,
                 isH2Downstream
-                    ? getH2HeadersAfterModification(reqUrl, clientReq.headers, modifiedReq?.headers)
-                    : { 'host': getHostAfterModification(reqUrl, clientReq.headers, modifiedReq?.headers) }
+                    ? getH2HeadersAfterModification(reqUrl, clientHeaders, modifiedReq?.headers)
+                    : !this.forwarding || !this.forwarding.updateHostHeader
+                        ? { 'host': getHostAfterModification(reqUrl, clientHeaders, modifiedReq?.headers) } 
+                        : {}
             );
```

Comment:
I think this isn't quite right, on a couple of points:

* In all cases, the same HTTP/1 logic should apply to HTTP/2 as well (which sets the `:authority` header instead of `host`, but it's exactly the same concept). If think the forwarding check should probably be an `if` around the whole `Object.assign`
* This treats `updateHostHeader` as defaulting to false (i.e. if it's undefined, it's false) but the default is `true` (unless you set `updateHostHeader: false` we always update the host header).

Can you update this to make those changes? I'll do some more detailed testing but other those points I think this is good to go :+1:

Reply:
Ah I see - updated in [3602ab4](https://github.com/httptoolkit/mockttp/pull/138/commits/3602ab4615d6825e492671e410fa81486386ac84)